gcloud/datastore/_implicit_environ.py

```diff
@@ -79,6 +86,16 @@ def compute_engine_id():
         connection.close()
 
 
+def _get_production_dataset_id():
```

Comment:
Yes they get monkeyed in `Test__determine_default_dataset_id._determine_default_helper`